### PR TITLE
#1 Add Plan

### PR DESCRIPTION
**Pull request by alexrudall**
- Add Plan CRUD
- Just start and end_time for now
-  Add user UI

### CODE REVIEW

File: app/models/plan.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#  updated_at         :datetime         not null
+#
+class Plan < ApplicationRecord
+  belongs_to :user, touch: true
+
+  validates :start_time, :end_time, :user_id, presence: true
+  validate :validate_end_time_after_start_time
+
+  def validate_end_time_after_start_time
+    return unless start_time && end_time && end_time < start_time
```
Comment (barbarawuk) [inline app/models/plan.rb:19]:
```suggestion
    return unless start_time && end_time && end_time <= start_time
```

Comment (barbarawuk) [reply app/models/plan.rb:19]: It's quite easy to create a plan with the default start and end which are the same, so perhaps worth validating they set a reasonable end time. (same goes for the time range validation)

**Review by barbarawuk — APPROVED** [review]
Nice 🙌 
One small validation suggestion.
Also I'm not sure about the user setting a start and end time, just the date would do?

**Comment by alexrudall** [comment]
Thanks for the review, rolling changes to this into [https://github.com/sardjv/murfin_method/pull/139](https://github.com/sardjv/murfin_method/pull/139)

**Pull request closed without merging**